cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -69,37 +69,31 @@
   private final DatasetOpExecutor opExecutorClient;
   private final ExploreFacade exploreFacade;
   private final boolean allowDatasetUncheckedUpgrade;
-  private final UsageRegistry usageRegistry;
```

Comment:
In the LineageWritingDatasetFramework. Added a call to registerUsage() in all the getDataset() methods there.